src/components/Form/Field.js

```diff
@@ -169,158 +112,144 @@ const ArrowDown = ({ size, fill, ...props }) => (
   </svg>
 )
 
-class Field extends Component {
-  constructor(props, context) {
-    super(props, context)
-    this.state = {
-      isFocused: false,
-      isValidating: false,
-      isDirty: false,
-      value: ''
-    }
-    this.inputRef = ref => (this.input = ref)
-  }
-  render() {
-    const {
-      onChange,
-      name,
-      autoComplete,
-      type,
-      simulate: sim,
-      label,
-      error,
-      renderInput,
-      onInc,
-      onDec,
-      icon,
-      disabled
-    } = this.props
+const Field = ({
+  onChange,
+  name,
+  autoComplete,
+  type,
+  simulate: sim,
+  label,
+  error,
+  renderInput,
+  onInc,
+  onDec,
+  icon,
+  disabled,
+  value
+}) => {
+  const [isFocused, setIsFocused] = useState(false)
+  const [isValidating, setIsValidating] = useState(false)
+  const [isDirty, setIsDirty] = useState(false)
+  const [localStateValue, setLocalStateValue] = useState('')
+  const inputRef = useRef()
+  const [colorScheme] = useColorContext()
 
-    let simulationClassName
-    let { isFocused } = this.state
+  useEffect(() => {
     if (sim) {
-      isFocused = sim.indexOf('focus') !== -1
-      simulationClassName = simulate(sim).toString()
+      setIsFocused(sim.indexOf('focus') !== -1)
     }
+  }, [sim])
+  const simulationClassName = useMemo(() => simulate(sim).toString(), [sim])
```

Comment:
```suggestion
  const simulationClassName = sim && simulate(sim).toString()
```

No `useMemo` necessary, especially because sim is only used in docs.